contxt/cli/commands/clusters.py

```diff
@@ -0,0 +1,120 @@
+from typing import List, Optional
+
+import click
+import yaml
+from requests.exceptions import HTTPError
+
+from contxt.cli.clients import Clients
+from contxt.cli.utils import fields_option, print_table, sort_option
+from contxt.models.contxt import Cluster
+
+
+def _get_org(clients: Clients, name: str):
+    organization = clients.contxt.get_organization_with_name(name)
+    if not organization:
+        raise click.ClickException(f"Organization {name!r} does not exist.")
+    return organization
+
+
+@click.group()
+def clusters() -> None:
+    """Clusters"""
+
+
+@clusters.command()
+@click.argument("cluster_slug", required=False)
+@click.option("--org", required=True, help="Org slug")
+@fields_option(default=["id", "host", "slug"], obj=Cluster)
+@sort_option(default="id")
+@click.pass_obj
+def get(clients: Clients, org: str, fields: List[str], sort: str, cluster_slug: Optional[str]) -> None:
+    """Get clusters"""
+    organization = _get_org(clients, org)
+    items = (
+        [clients.contxt_deployments.get_cluster(organization.id, cluster_slug)]
+        if cluster_slug
+        else clients.contxt_deployments.get_clusters(organization.id)
+    )
+    print_table(items=items, keys=fields, sort_by=sort)
+
+
+@clusters.command()
+@click.argument("host")
+@click.option("--org", required=True, help="Org slug")
+@click.pass_obj
+def login(clients: Clients, host: str, org: str) -> None:
+    """Get clusters"""
+    try:
+        token = clients.auth.get_token_provider(audience=host).access_token
+        organization = _get_org(clients, org)
+        clusters = clients.contxt_deployments.get_clusters(organization.id)
+        cluster = next((cluster.slug for cluster in clusters if cluster.host == host), None)
+        kubeconfig = {
+            "kind": "Config",
+            "apiVersion": "v1",
+            "preferences": {},
+            "current-context": cluster,
+            "users": [{"name": cluster, "user": {"token": token}}],
+            "clusters": [{"name": cluster, "cluster": {"server": host}}],
+            "contexts": [
+                {
+                    "name": cluster,
+                    "context": {"cluster": cluster, "namespace": "default", "user": cluster},
+                }
+            ],
+        }
+        print(yaml.safe_dump(kubeconfig))
+    except HTTPError:
+        pass
+
+
+@clusters.command()
+@click.argument("host")
+@click.option("--org", required=True, help="Org slug")
+@click.option(
+    "--description",
+    required=True,
+    help="Information about what this cluster is for and " "what environment it belongs to",
+)
+@click.option(
+    "--infrastructure-id",
+    required=True,
+    help="The ID of the infrastructure registered in "
+    "our system. Ask ndustrial.io DevOps "
+    "what this value should be",
+)
```

Comment:
i don't think there's a way to know what valid values there are for this, right?

Reply:
The api will return if it's not valid